app/lib/month.rb

```diff
@@ -66,9 +66,12 @@ def self.current
   def self.parse_month(month)
     return month if month.is_a?(Month)
 
-    month_params = month.split('-').first(2)
-    Month.new(*month_params)
-  rescue ArgumentError, NoMethodError
+    match = /^(\d{4})-(\d{2})(?:-\d{2})?$/.match(month)
+
+    raise ArgumentError unless match
+
+    Month.new(*match.captures)
+  rescue ArgumentError, NoMethodError, RangeError
```

Comment:
Why not `StandardError`? That way the user will never see a `500`. What are the consequences of returning `nil`?

Reply:
Dunno... just trying to be more specific :)

I guess it makes sense, as we just return a `nil` value in case anything went wrong with Month initialization. That's certainly better than keeping an unhandled exception.

I've checked and there are two main uses of the `Month.parse_month`: 
- when setting the invoice period (setting a `nil` will result in a validation error)
- in `Invoice.by_month` scope https://github.com/3scale/porta/blob/fix-invoice-period-validation/app/models/invoice.rb#L72  There should be no `nil` values for period, and I think we're not calling this scope with user-provided params, so should be OK.